insights/parsers/tests/test_lssap.py

```diff
@@ -0,0 +1,133 @@
+from ...parsers import lssap, ParseException
+from ...tests import context_wrap
+from ...util import keys_in
+import pytest
+
+Lssap_BAD = """
+ - lssap version 1.0 -
+ ==========================================
+   SID   Nr   Instance    SAPLOCALHOST                        Version                 DIR_EXECUTABLE
+   HA2|  16|       D16|         lu0417
+""".strip()
+
+Lssap_BAD1 = """
+   HA2|  16|       D16|         lu0417
+""".strip()
+
+Lssap_BAD2 = ""
+
+Lssap_nw_TEST = """
+ - lssap version 1.0 -
+==========================================
+  SID   Nr   Instance    SAPLOCALHOST                        Version                 DIR_EXECUTABLE
+  HA2|  16|       D16|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/D16/exe
+  HA2|  22|       D22|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/D22/exe
+  HA2|  50|       D50|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/D50/exe
+  HA2|  51|       D51|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/D51/exe
+  HA2|  52|       D52|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/D52/exe
+""".strip()
+
+Lssap_hana_TEST = """
+ - lssap version 1.0 -
+==========================================
+  SID   Nr   Instance    SAPLOCALHOST                        Version                 DIR_EXECUTABLE
+  HA2|  16|       HDB16|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/HDB16/exe
+""".strip()
+
+Lssap_ascs_TEST = """
+ - lssap version 1.0 -
+==========================================
+  SID   Nr   Instance    SAPLOCALHOST                        Version                 DIR_EXECUTABLE
+  HA2|  16|       ASCS16|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/ASCS16/exe
+""".strip()
+
+Lssap_runAll_Test = """
+ - lssap version 1.0 -
+==========================================
+  SID   Nr   Instance    SAPLOCALHOST                        Version                 DIR_EXECUTABLE
+  HA2|  16|       D16|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/D16/exe
```

Comment:
Is this really the way this output would be formatted?  Or have you constructed it by hand?  It'd be really good if we could get real data.

Reply:
@PaulWay This is exactly how the output looks. The SAP experts I'm working with sent me doc with the data in it and expressed that this is what the output looks like. I don't have a machine for myself to test with yet.

Reply:
I was more remarking on the odd semi-justification and names being longer and shorter - sorry to have doubted you @SteveHNH :-)

Reply:
@PaulWay No worries! I got some test machines finally and the output was accruate. I did have to add a `=======` at the bottom as well, but that was the only modification .